airflow/utils/context.py

```diff
@@ -281,10 +289,13 @@ def __getitem__(self, key: int | str | Asset | AssetAlias) -> LazyAssetEventSele
             asset_alias = self._asset_aliases[obj.name]
             join_clause = AssetEvent.source_aliases
             where_clause = AssetAliasModel.name == asset_alias.name
-        elif isinstance(obj, (Asset, str)):
+        elif isinstance(obj, (Asset, AssetRef)):
+            join_clause = AssetEvent.asset
+            where_clause = AssetModel.name == self._assets[obj.name].name
+        elif isinstance(obj, str):
             asset = self._assets[extract_event_key(obj)]
             join_clause = AssetEvent.asset
-            where_clause = AssetModel.uri == asset.uri
+            where_clause = AssetModel.name == asset.name
```

Comment:
Hmm… I wonder if we should try to support _both_ name and URI here. Also this current code would not work because `extract_event_key` returns the URI.

Reply:
Right. This one is a bit tricky and intended to keep it this way for now. (This supports assets like `Asset(uri="...")` as the name fallbacks to its uri)

Reply:
I'm actually thinking of removing `str` support entirely. As we support more and more types, passing a string could be pretty confusing.

Reply:
That’s true. We’ll need to deprecate it in 2 though.

Reply:
Sounds good to me

Reply:
PR for 1 created